### DISCUSSION TITLE
Unable to make smart contract call to load data with v2

### DISCUSSION THREAD

**Discussion by mcoetzee** [WalletConnect/walletconnect-monorepo#2694] [Developer Support]
### Repo to replicate the issue
https://github.com/mcoetzee/wallet-connect-bug

### What is your setup?
I am using the latest 
```
"wagmi": "^1.2.0",
"@web3modal/ethereum": "^2.4.7",
"@web3modal/react": "^2.4.7",
"viem": "^1.0.7",
```
with the Ethers.js adapters provided by `wagmi` (https://wagmi.sh/core/ethers-adapters)

### What is happening?
When constructing a smart contract with a `signer`, it fails when calling a method on the smart contract that loads data from the chain. It works as expected with Wallet Connect v1, but fails when I use Wallet Connect v2.
```ts
    const contract = new ethers.Contract(
      "0x...",
      someAbi,
      signer
    );

    await contract.getSomeData(); // This throws an error
```
`
Error: missing revert data in call exception; Transaction reverted without a reason string [ See: https://links.ethers.org/v5-errors-CALL_EXCEPTION 
`

### Steps to reproduce the issue
`yarn && yarn start` the code at https://github.com/mcoetzee/wallet-connect-bug
1. Connect with MetaMask (browser extension) to see the data load as expected
2. Disconnect and then connect a Trust/Rainbow Wallet (by scanning QR code) to see the error when loading data

**Comment by glitch-txs** [comment]
Hey, for read calls you can make it work by constructing the contract instance with the provider instead of the signer, for gas transaction you can use the signer
